### PR TITLE
Adding File Support

### PR DESCRIPTION
**Pull request by splintercat**
Adding basic support for files in Cerberus.

Uploading a file follows the same conventions as the requests library where you pass a file handle of a file opened in binary mode to the function.

Right now no checks are made to ensure that the uploaded file size is under 256k.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,6 +147,63 @@ Categories are for organizing safe deposit boxes.  To list the available categor
...
+## Returns
+b'example file. With binary data \xAB\xBA\xCA\xB0'
+```
+
+To download just the file metadata
+```python
+client.get_file_metadata('category/sdb/path/to/file.example')
+
+## Returns
+{'Date': 'Thu, 10 May 2018 00:57:00 GMT', 'Content-Type': 'application/octet-stream; charset=UTF-8', 'Content-Length': '36', 'Connection': 'keep-alive', 'Content-Disposition': 'attachment; filename="test.py"', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains', 'X-B3-TraceId': 'beede324324324'}
```
Comment (sdford) [inline README.md]: Same here about pretty print

File: cerberus/client.py [hunk 2]
```diff
@@ -324,6 +324,169 @@ def update_sdb(self, sdb_id, owner=None, description=None, user_group_permission
...
+
+        payload = {'versionId': str(version)}
+        secret_resp = requests.head(str.join('', [self.cerberus_url, '/v1/secure-file/', vault_path]),
+                                    params=payload, headers=self.HEADERS)
+
+        throw_if_bad_response(secret_resp)
+
+        return secret_resp.headers
+
+    def _get_files(self, vault_path, version=None):
```
Comment (sdford) [inline cerberus/client.py]: Should this be `_get_file` singular?

File: README.md [hunk 3]
```diff
@@ -147,6 +147,63 @@ Categories are for organizing safe deposit boxes.  To list the available categor
 client.get_categories()
 ```
 
+
+#### Read files from Cerberus
```
Comment (sdford) [inline README.md]: Please add `list_files` here as well

**Comment by splintercat** [comment]
Doing the json formatting is pretty bad.
I ran the output through python and used it's pretty printing to fix it up.

File: cerberus/client.py [hunk 4]
```diff
@@ -324,6 +324,169 @@ def update_sdb(self, sdb_id, owner=None, description=None, user_group_permission
...
+
+        # Because of the addition of versionId and the way URLs are constructed, secure_data_path should
+        #  always end in a '/'.
+        secure_data_path = self._add_slash(secure_data_path)
+        secret_resp = requests.get(self.cerberus_url + '/v1/secure-files/' + secure_data_path,
+                                   params=payload, headers=self.HEADERS)
+        throw_if_bad_response(secret_resp)
+        return secret_resp.json()
+
+    def put_file(self, secure_data_path, filename, filehandle, content_type=None):
```
Comment (mayitbeegh) [inline cerberus/client.py]: This function requires file names in two places: path and filename. Is there a reason to having this duplication?

Comment (splintercat) [reply cerberus/client.py]: From reading the API docs, it looks like filename is required as part of the metadata and isn't derived from the path. 

Form a quick test where I omit the filename, Cerberus will throw an error.  (Granted it could also be my code doing something wrong when it makes the request).

I also tested just suppling the path without the filename, and then supplying the filename in the metadata, but that also didn't work. (once again, it could be my code doing the wrong thing).

Basically it looks like the filename in the path is just the path, whereas the filename in the header information is supplied by the uploader. 

I could try and write the code so that is takes a path without a filename, and then uses the filename parameter to complete the path for you.  But I feel that might be confusing since writing secrets into Cerberus follows a similar  kind of pattern.

Comment (sdford) [reply cerberus/client.py]: @splintercat That is partially correct. The filename is required in the HTTP form multipart, but is ignored by Cerberus. What matters is the filename that is after the last slash in the path. I realize the API docs may be misleading in this regard, so let me know if you have ideas on how they can be made more clear.

I think the best way to avoid confusion is to ensure that the full path (including the filename, at the end) is provided to the `put_file` method. Then the filename can be parsed from the given path.

Let me know what you think.

Comment (splintercat) [reply cerberus/client.py]: I can see about adding that in. 
It does make sense that those two parts should match.
